docs/contributing.md

```diff
@@ -0,0 +1,115 @@
+
+#### As a MetaMask contributor
+
+If you're a MetaMask engineer, you can create these preview versions via draft pull requests:
+
+1. Navigate to your settings within GitHub and [create a classic access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). Make sure to give this token the `packages:read` scope.
+2. Switch to your project locally and add a `.npmrc` file with the following content, filling in the appropriate areas:
+   ```
+   @metamask:registry=https://npm.pkg.github.com
+   //npm.pkg.github.com/:_authToken=<your personal access token>
+   ```
+   Make sure not to commit this file.
+3. Go to GitHub and open up a pull request for this repository (for the extension or mobile app, this needs to be a draft PR), then post a comment on the PR with the text `@metamask-bot preview-build`. (This triggers the `publish-preview` GitHub action.)
+4. After a few minutes, you will see a new comment indicating that all packages have been published with the format `<package name>-<commit id>`.
+5. Switch back to your project locally and update `package.json` by replacing the versions for the packages you've changed in your PR using the new version format (e.g. `1.2.3-e2df9b4` instead of `~1.2.3`), then run `yarn install`.
+6. Repeat steps 3-5 after pushing new changes to your PR to generate and use new preview versions.
+
+#### As an independent contributor
+
+If you're a contributor and you've forked this repository, you can create preview versions for a branch (or any arbitrary commit) via scripts:
+
+1. Navigate to your settings within GitHub and [create a **classic** access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). Make sure to give this token the `packages:read` scope.
+2. Switch to your project locally and add a `.npmrc` file with the following content, filling in the appropriate areas:
+   ```
+   @<your GitHub username>:registry=https://npm.pkg.github.com
+   //npm.pkg.github.com/:_authToken=<your personal access token>
+   ```
+   Make sure not to commit this file.
+3. Open the `package.json` for each package that you want to publish and change the scope in the name from `@metamask` to `@<your GitHub username>`.
+4. Switch to your fork of this repository locally and run `commit_sha=$(git rev-parse --short HEAD)`. This generates the identifier appended to all package versions to associate them with the current commit. (Alternatively, you can set `commit_sha` to any arbitrary commit id.)
```

Comment:
Nit: Could we skip this step, and use `yarn prepare-preview-builds "$(git rev-parse --short HEAD)"` in the next step instead? Better one command than two. I don't think we should be encouraging people to choose any arbitrary id either, that could be confusing (it will always use the local tree to publish).

I see that it is referenced later as well, but it should be easy enough for users to see what the preview build numbers are (it's in the CLI output, and users can always check the package manifests for reference).

Reply:
> it should be easy enough for users to see what the preview build numbers are (it's in the CLI output, and users can always check the package manifests for reference)

It's technically in the CLI output, but it's not very clear IMO. I guess we can improve this developer experience in another PR though.

Reply:
Updated in [bae8925](https://github.com/MetaMask/controllers/pull/993/commits/bae89258a3e9936fa0f6b3c4214a4f011cadd749).

Reply:
Agreed, definitely could be made more user friendly